pkg/kubectl/service.go

```diff
@@ -183,7 +183,12 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 	if len(params["type"]) != 0 {
 		service.Spec.Type = api.ServiceType(params["type"])
 	}
-	if service.Spec.Type == api.ServiceTypeLoadBalancer {
+	switch service.Spec.Type {
+	case api.ServiceTypeNodePort:
+		for i, port := range service.Spec.Ports {
+			service.Spec.Ports[i].NodePort = port.Port
```

Comment:
This is both semantically wrong (nodePort != port) and behaviorally wrong (you are assigning the same value to to potentially many ports, which obviously can not work).